spec/lib/content_exporter_spec.rb

```diff
@@ -0,0 +1,199 @@
+        show_to_nations: "Wales OR Scotland OR England",
+        subgroup_title: "I am the title for Group one subgroup one",
+        support_and_advice: true,
+        text: "This is a row has text, an href and multiple group criteira, it will appear as an anchor tag if the user's answers match the more complex criteria" },
+      { id: "0005",
+        group_title: "I am the title for Group one",
+        href: "",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group one subgroup two",
+        support_and_advice: false,
+        text: "This is a row that only has text, it will appaer like a paragraph" },
+      { id: "0006",
+        group_title: "I am the title for Group two",
+        href: "",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group two subgroup one",
+        support_and_advice: false,
+        text: "This is a row that only has text, it will appaer like a paragraph" },
+    ]
+
+    before do
+      allow(I18n).to receive(:t).with("results_link") { locale_results_link_fixture }
+      allow(I18n).to receive(:t).with("coronavirus_form.groups") { locale_form_groups_fixture }
+    end
+
+    it "returns an array of result link rows" do
+      expect(results_rows).to eq(results_row_fixture)
+    end
+
+    it "returns an empty string value when a link doesn't have :show_to_nations criteria" do
+      expect(results_rows[0][:show_to_nations]).to eq("")
+    end
+
+    it "returns a single string when a link has a single :show_to_nations criteira" do
+      expect(results_rows[2][:show_to_nations]).to eq("Wales")
+    end
+
+    it "returns a singe string with joined with OR when a link has multiple :show_to_nations criteira" do
+      expect(results_rows[3][:show_to_nations]).to eq("Wales OR Scotland OR England")
+    end
+
+    it "returns a href URL string when a link has a href value" do
+      expect(results_rows[1][:href]).to eq("http://test.stubbed.gov.uk")
+    end
+
+    it "returns an empty string when a link has no href value" do
+      expect(results_rows[0][:href]).to eq("")
+    end
+
+    it "returns a heading string when a link has a group_title value" do
+      expect(results_rows[0][:group_title]).to eq("I am the title for Group one")
+    end
+
+    it "returns a heading string when a link has a subgroup_title value" do
+      expect(results_rows[0][:subgroup_title]).to eq("I am the title for Group one subgroup one")
+    end
+
+    it "returns a text string when a link has a text value" do
+      expect(results_rows[0][:text]).to eq("This is a row that only has text, it will appaer like a paragraph")
+    end
+
+    it "returns a boolean support_and_advice true when the link is under support_and_advice" do
+      expect(results_rows[3][:support_and_advice]).to be(true)
+    end
+
+    it "returns a boolean support_and_advice false when the link is under items" do
+      expect(results_rows[0][:support_and_advice]).to be(false)
+    end
```

Comment:
These tests are redundant given the test on line 42, where the equality of the received array is checked.

The tests, or rather the test descriptions on these lines, do add context however. As functionally they are commentary; might it make more sense to just make them be comments on/in the fixture definition, stating why each line is included?

(Note - there isn't a strong case for change here; overly verbose tests aren't really a problem, and I'm happy to be pushed back on this!)

Reply:
Yeah fair comment, I guess i'm worried that there's a lot going on there and wanted to follow a "Tests as documentation" method.

I was thinking, downside is probably that redundant tests slow down test suites and that slows our deploys... but the extra time here seems tiny. And i suppose you'd have to update two expecations instead of one if we made a change...

Humm, i suppose as someone coming to this afresh, did having them there help you understand what I was trying to do? If so i'd rather keep em. If not i'll remove them?

Comments could work too, but if there's minimal cost to having these i might leave them as they are for now.

Reply:
I'm quite on the fence tbh - they certainly do no harm! 🙂